pkg/ratelimit/limitware/middleware.go

```diff
@@ -197,94 +196,88 @@ func (m *Middleware) Handle(next http.Handler) http.Handler {
 }
 
 // APIKeyFunc returns a default key function for ratelimiting on our API key
-// header. It falls back to rate limiting by the client ip.
-func APIKeyFunc(ctx context.Context, scope string, db *database.Database) httplimit.KeyFunc {
+// header. Since APIKeys are assumed to be "public" at some point, they are
+// rate limited by [api-key,ip].
+func APIKeyFunc(ctx context.Context, scope string, hmacKey []byte) httplimit.KeyFunc {
```

Comment:
we still need to put this in a realm level bucket and limit the number of verification checks on the realm over a period of time.

the idea was that we don't one one or a small set of IPs to eat all the quota for the realm.

Reply:
Done